src/js/TuningSliders.js

```diff
@@ -0,0 +1,313 @@
+    PID_DEFAULT: [],
+    FILTER_DEFAULT: {},
+};
+
+TuningSliders.initialize = function() {
+    this.PID_DEFAULT = FC.getPidDefaults();
+    this.FILTER_DEFAULT = FC.getFilterDefaults();
+
+    this.setDMinFeatureEnabled($('#dMinSwitch').is(':checked'));
+
+    this.initPidSlidersPosition();
+    this.initGyroFilterSliderPosition();
+    this.initDTermFilterSliderPosition();
+
+    this.updatePidSlidersDisplay();
+    this.updateFilterSlidersDisplay();
+}
+
+TuningSliders.setDMinFeatureEnabled = function(dMinFeatureEnabled) {
+    this.dMinFeatureEnabled = dMinFeatureEnabled;
+    if (this.dMinFeatureEnabled) {
+        this.defaultPDRatio = this.PID_DEFAULT[0] / this.PID_DEFAULT[2];
+    } else {
+        this.defaultPDRatio = this.PID_DEFAULT[0] / this.PID_DEFAULT[3];
+    }
+}
+
+TuningSliders.scaleSliderValue = function(value) {
+    if (value > 1) {
+        return Math.round(((value - 1) * 2 + 1) * 10) / 10;
+    } else {
+        return value;
+    }
+};
+
+TuningSliders.updatePidSlidersDisplay = function() {
+    // check if pid values changed manually by saving current values, doing the slider based calculation, and comaparing
+    // if values before and after calculation, if all of them are equal the values haven't been changed manually
+    // and the sliders are shown, otherwise sliders are grayed out and centered
+    this.pidSlidersUnavailable = false;
+    let currentPIDs = [];
+    PID_names.forEach(function(elementPid, indexPid) {
+        let searchRow = $('.pid_tuning .' + elementPid + ' input');
+        searchRow.each(function (indexInput) {
+            if (indexPid < 3 && indexInput < 5) {
+                currentPIDs.push($(this).val());
+            }
+        });
+    });
+    this.calculateNewPids();
+    PID_names.forEach(function(elementPid, indexPid) {
+        let searchRow = $('.pid_tuning .' + elementPid + ' input');
+        searchRow.each(function (indexInput) {
+            if (indexPid < 3 && indexInput < 5) {
+                if (currentPIDs[indexPid * 5 + indexInput] != $(this).val()) {
+                    TuningSliders.pidSlidersUnavailable = true;
+                }
+                $(this).val(currentPIDs[indexPid * 5 + indexInput]);
+            }
+        });
+    });
+    if ($('input[id="useIntegratedYaw"]').is(':checked')) {
+        this.pidSlidersUnavailable = true;
+    }
+    if (this.pidSlidersUnavailable) {
+        $('.tuningPIDSliders').css('opacity', 0.3);
+        $('.tuningPIDSliders output').hide();
+        $('.slidersDisabled').show()
+        $('#tuningMasterSlider, #tuningPDRatioSlider, #tuningPDGainSlider, #tuningResponseSlider').val(1);
+    } else {
+        $('.tuningPIDSliders').css('opacity', '');
+        $('.tuningPIDSliders output').show();
+        $('.slidersDisabled').hide();
+        this.initPidSlidersPosition();
+    }
+    if ((PIDs[1][0] > 110 || PIDs[1][1] > 120 || PIDs[1][2] > 70 || ADVANCED_TUNING.dMinPitch > 40) && !this.pidSlidersUnavailable) {
```

Comment:
This values are better defined like constants or similar.

Reply:
Yes, will change that.